src/openxr/OpenXRApi.cpp

```diff
@@ -2896,7 +2914,21 @@ bool OpenXRApi::get_head_center(float world_scale, godot_transform *transform) {
 		.type = XR_TYPE_SPACE_LOCATION,
 		.next = NULL
 	};
-	result = xrLocateSpace(view_space, play_space, frameState.predictedDisplayTime, &location);
+
+	XrTime time;
+	if (form_factor == XR_FORM_FACTOR_HANDHELD_DISPLAY) {
+		// For handheld displays we are rendering MONO and this method is called both for our rendering position
+		// and to position our camera node for the next frame.
+		// It is more important to get the rendering right here.
+		// Note, there currently are no platforms where OpenXR is used with Godot that support this mode
+		// so it's probably a non issue for the time being. This is already resolved in Godot 4.
+		time = frameState.predictedDisplayTime;
+	} else {
```

Comment:
Would this logic need to be repeated in other locations (e.g: `action.cpp` or `xr_ext_hand_tracking_extension_wrapper.cpp`)? 
If so, you could move this logic in `get_next_frame_time` (would need to be renamed adequately) as well.

Reply:
No, this is purely a problem with this function because Godot can call it when it is retrieving the transform with which it renders. That is the only transform that requires the current frame time.

This is only a problem with mono setups like handhelds (which we don't support). With Stereo this method is only called when we update the ARVRCamera nodes position for which we do want the next frames timing. For stereo rendering it calls 'get_view_transform' to get the transform for the individual eyes.

I've already changed the logic for Godot 4, I don't think it's worth worrying about for Godot 3.